src/cdk/schematics/ng-update/upgrade-data.ts

```diff
@@ -68,5 +68,6 @@ export interface RuleUpgradeData {
 export function
 getVersionUpgradeData<T extends keyof RuleUpgradeData, U = ValueOfChanges<RuleUpgradeData[T]>>(
     r: MigrationRule<RuleUpgradeData>, dataName: T): U[] {
-  return getChangesForTarget<U>(r.targetVersion, r.upgradeData[dataName] as VersionChanges<U>);
+  return getChangesForTarget<U>(r.targetVersion,
+                                r.upgradeData[dataName] as unknown as VersionChanges<U>);
```

Comment:
Why do we need to cast to `unknown` here? The `upgradeData` should be typed with `RuleUpgradeData` and have properties that always resolve to `VersionChanges`?

Reply:
TS 3.5 is a bit more strict about generic types. I was getting this:

```
Conversion of type 'RuleUpgradeData[T]' to type 'VersionChanges<U>' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
```

Reply:
hmm. it looks like `T` is somehow not properly interpreted as a key of `RuleUpgradeData`. I'm fine with leaving it like that, but can we add a super-small comment for this?

Reply:
Done.